src/core/solvers/ipopt/ipopt.cpp

```diff
@@ -0,0 +1,48 @@
+#include "crocoddyl/core/solvers/ipopt/ipopt.hpp"
+
+namespace crocoddyl {
+
+SolverIpOpt::SolverIpOpt(boost::shared_ptr<crocoddyl::ShootingProblem> problem)
+    : SolverAbstract(problem), ms_nlp_(new MultipleShootingNlp(problem)), app_(IpoptApplicationFactory()) {
+  app_->Options()->SetNumericValue("tol", 3.82e-6);
+  app_->Options()->SetStringValue("mu_strategy", "adaptive");
+  // app_->Options()->SetStringValue("max_iter", 100);
+  // app->Options()->SetStringValue("output_file", "ipopt.out");
+
+  status_ = app_->Initialize();
+
+  if (status_ != Ipopt::Solve_Succeeded) {
+    // Throw an exception here
+    std::cout << std::endl << std::endl << "*** Error during initialization!" << std::endl;
+  }
+}
+
+bool SolverIpOpt::solve(const std::vector<Eigen::VectorXd>& init_xs, const std::vector<Eigen::VectorXd>& init_us,
+                        const std::size_t maxiter, const bool is_feasible, const double regInit) {
+  assert(init_xs.size() == ms_nlp_->get_problem()->get_T() + 1);
+  assert(init_us.size() == ms_nlp_->get_problem()->get_T());
+
+  if (init_xs != DEFAULT_VECTOR) {
+    ms_nlp_->set_xs(init_xs);
+  }
+  if (init_us != DEFAULT_VECTOR) {
+    ms_nlp_->set_us(init_us);
+  }
```

Comment:
Use  `setCandidate`, i.e.,
```suggestion
setCandidate(init_xs, init_us, is_feasible);
ipopt_iface_->set_xs(xs_);
ipopt_iface_->set_us(us_);
```